android/capacitor/src/main/java/com/getcapacitor/plugin/PushNotifications.java

```diff
@@ -161,13 +162,27 @@ public static void sendRemoteMessage(RemoteMessage remoteMessage) {
   }
 
   public void fireNotification(RemoteMessage remoteMessage) {
-    Map<String, Object> data = new HashMap<String, Object>();
+    JSObject remoteMessageData = new JSObject();
+
+    JSObject data = new JSObject();
     for (String key : remoteMessage.getData().keySet()) {
       Object value = remoteMessage.getData().get(key);
       data.put(key, value);
     }
-    JSObject remoteMessageData = new JSObject();
     remoteMessageData.put("data", data);
+
+    RemoteMessage.Notification notification = remoteMessage.getNotification();
+    if (notification != null) {
+      Uri link = notification.getLink();
```

Comment:
Can this be a String instead of Uri?

Reply:
I'm a bit rusty with Java so I don't know the best practice here. getLink returns a Uri object or null. Is there a shorthand way of getting the String value of the Uri or null?

Reply:
It has a .toString() that should work, but didn’t test with an actual link

Reply:
But getLink can return null, which would throw an exception if we called toString there

Reply:
Sorry, I didn’t notice you are already doing the toString later, so that’s enough